src/pages/index.js

```diff
@@ -62,25 +60,140 @@ popupWithImage.setEventListeners();
 const userInfo = new UserInfo({
   nameSelector: ".profile__title",
   jobSelector: ".profile__description",
+  avatarSelector: ".profile__image",
+});
+
+//Api
+
+const api = new Api({
+  baseUrl: "https://around-api.en.tripleten-services.com/v1",
+  headers: {
+    authorization: "8fb04c52-2fa8-4c47-a8b4-76fb0eacbed1",
+    "Content-Type": "application/json",
+  },
 });
+//userinfome
+api
+  .getUserInfo()
+  .then((data) => {
+    console.log(data);
+    userInfo.setUserInfo(data);
+  })
+  .catch((err) => {
+    console.error(err);
+  });
 
 /*Functions*/
 
+const handleAvatarSubmit = ({ avatar }) => {
+  console.log(avatar);
+  editAvatarPopup.setLoading(true);
+  api
+    .updateAvatar(avatar)
+    .then((data) => {
+      userInfo.setUserInfo(data);
+      formValidators["edit-avatar-form"].disableButton();
+      formValidators["edit-avatar-form"].toggleButtonState();
+      editAvatarPopup.close();
+    })
+    .catch((err) => {
+      console.error(err);
+    })
+    .finally(() => {
+      editAvatarPopup.setLoading(false);
+    });
+};
+
+const editAvatarPopup = new PopupWithForm(
+  "#edit-avatar-modal",
+  handleAvatarSubmit
+);
+editAvatarPopup.setEventListeners();
+
+//edit profile
 function handleProfileEditSubmit({ name, description }) {
-  userInfo.setUserInfo({ name, description });
-  profilePopup.close();
+  profilePopup.setLoading(true);
+  api
+    .updateProfileInfo(name, description)
+    .then((userData) => {
+      console.log(userData);
+      userInfo.setUserInfo(userData);
+      profilePopup.close();
+    })
+    .catch((err) => {
+      console.error(err);
+    })
+    .finally(() => {
+      profilePopup.setLoading(false);
+    });
 }
 
 function handleImageClick(cardData) {
   popupWithImage.open(cardData);
 }
 
-function handleAddCardFormSubmit(e) {
-  const cardTitleInput = addCardFormElement.querySelector("#name");
-  const cardUrlInput = addCardFormElement.querySelector("#url");
-  const name = cardTitleInput.value;
-  const link = cardUrlInput.value;
-  renderCard({ name, link });
+//add card
+function handleAddCardFormSubmit({ name, link }) {
+  cardPopup.setLoading(true);
+  api
+    .createNewCard({ name, link })
+    .then((data) => {
+      const cardElement = createCard(data);
+      cardSection.addItem(cardElement);
+      cardPopup.close();
+    })
+    .catch((err) => {
+      console.error(err);
+    })
+    .finally(() => {
+      cardPopup.setLoading(false);
+    });
+}
+
+//deletecard
+function handleDeleteButton(card) {
+  deletePopup.open();
+  deletePopup.setSubmit(() => {
+    console.log(card);
```

Comment:
Needs correction: don't forget to notify the user about loading by changing the confirm button text